Lib/test/test_mailcap.py

```diff
@@ -245,6 +241,19 @@ def test_test(self):
         ]
         self._run_cases(cases)
 
+    def test_unsafe_mailcap_input(self):
+        c     = MAILCAPDICT
+        plist = ["total=*"]
+
+        with self.assertWarnsRegex(mailcap.UnsafeMailcapInput,'Refusing to substitute parameter.*into a shell command'):
+            self.assertEqual(mailcap.subst("echo %{total}", "audio/wav", "foo.txt", plist), None)
+        
+        with self.assertWarnsRegex(mailcap.UnsafeMailcapInput,'Refusing to substitute MIME type.*into a shell'):
+            self.assertEqual(mailcap.subst("echo %t", "audio/*", "foo.txt"), None)
+        
+        with self.assertWarnsRegex(mailcap.UnsafeMailcapInput,'Refusing to use mailcap with filename.*Use a safe temporary filename.'):
```

Comment:
PEP 8: wrap lines at 80 chars, and have whitespace between the `,` and the next function arg. 


```suggestion
        with self.assertWarnsRegex(mailcap.UnsafeMailcapInput,
                                                         'Refusing to use mailcap with filename.*Use a safe temporary filename.'):
```